src/rsz/src/Resizer.cc

```diff
@@ -264,7 +266,70 @@ void Resizer::removeBuffers()
   logger_->info(RSZ, 26, "Removed {} buffers.", remove_count);
 }
 
-bool Resizer::bufferBetweenPorts(Instance* buffer)
+LibertyCell *Resizer::findSmallestEquiv(const DcalcAnalysisPt *dcalc_ap,
+                                        LibertyCell *cell)
+{
+  LibertyCellSeq *equiv_cells = sta_->equivCells(cell);
+  if (!equiv_cells)
+    return cell;
+
+  LibertyPort *out_port = nullptr;
+  sta::LibertyCellPortIterator it(cell);
+  while (it.hasNext()) {
+    LibertyPort *port = it.next();
+    if (port->direction()->isOutput()) {
+      if (out_port)
+        return cell;
+      out_port = port;
+    }
+  }
+
+  if (!out_port)
+    return cell;
+
+  sort(equiv_cells,
+       [=] (const LibertyCell *cell1, const LibertyCell *cell2) {
+          LibertyPort *port1 = cell1->findLibertyPort(out_port->name());
+          LibertyPort *port2 = cell2->findLibertyPort(out_port->name());
+          float drive1 = port1->driveResistance();
+          float drive2 = port2->driveResistance();
+          ArcDelay intrinsic1 = port1->intrinsicDelay(this);
+          ArcDelay intrinsic2 = port2->intrinsicDelay(this);
+          return (std::tie(drive1, intrinsic2) < std::tie(drive2, intrinsic1));
+       });
+
+  return equiv_cells->back();
+}
+
+void
+Resizer::downsizeAllCells()
```

Comment:
What is the use model for this?

Reply:
Buffering interacts with sizing so it's a potential debugging aid. Also we can test if for some designs, e.g. with multiple clock domains, it might be better to start with a clean slate with regards to sizing since ABC wasn't aware of clock domains.

I wrote this command first because it sounded like a simple thing I could finish end-to-end and familiarize myself with the STA/OR interfaces on it.